src/k8s-engine/lima.ts

```diff
@@ -1096,14 +1104,45 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
       await this.ssh('mkdir', '-p', 'bin');
       await this.lima('copy', scriptPath, `${ MACHINE_NAME }:bin/install-k3s`);
       await this.ssh('chmod', 'a+x', 'bin/install-k3s');
-      await fs.promises.chmod(path.join(paths.cache, 'k3s', version.raw, k3s), 0o755);
-      await this.ssh('sudo', 'bin/install-k3s', version.raw, path.join(paths.cache, 'k3s'));
+      if (this.#enabledK3s) {
+        await fs.promises.chmod(path.join(paths.cache, 'k3s', version.raw, k3s), 0o755);
+        await this.ssh('sudo', 'bin/install-k3s', version.raw, path.join(paths.cache, 'k3s'));
+      }
       await this.lima('copy', resources.get('scripts', 'profile'), `${ MACHINE_NAME }:~/.profile`);
     } finally {
       await fs.promises.rm(workdir, { recursive: true });
     }
   }
 
+  protected async configureContainerd(): Promise<void> {
+    const workdir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-containerd-install-'));
+
+    try {
+      const fixedProfile = path.join(workdir, 'profile');
+      const profileContents = (await fs.promises.readFile(resources.get('scripts', 'profile'))).toString();
+      const fixedContents = profileContents.replace(/export CONTAINERD_ADDRESS=.*/,
+        `export CONTAINERD_ADDRESS=${ CONTAINERD_ADDRESS_K3S }`);
+      const confListPath = path.join(workdir, '10-flannel.conflist');
+      const configPath = path.join(workdir, 'config.toml');
+
+      await fs.promises.writeFile(fixedProfile, fixedContents);
+      await this.lima('copy', fixedProfile, `${ MACHINE_NAME }:~/.profile`);
```

Comment:
Might be good to adjust `this.writeFile` to be able to write files as non-root.

Reply:
I don't think we need ~/.profile -- so this is incentive to remove it now.

Reply:
And all the calls to `writeFile` are writing files into `/etc/{conf.d,init.d}`, so no need for that. The `.profile` is created with default ownership, not root.